src/preprocess/acdc_data_split.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+import csv
+import logging
+import random
+from collections.abc import Iterable
+from pathlib import Path
+
+
+def main(args):
+    # Randomly and evenly split the data into k folds based on the diseases.
+    data_paths = sorted((args.data_dir / 'training').glob('**/Info.cfg'))
+    group_dict = {
+        'NOR': [],  # healthy patients
+        'MINF': [],  # patients with previous myocardial infarction
+        'DCM': [],  # patients with dilated cardiomyopathy
+        'HCM': [],  # patients with an hypertrophic cardiomyopathy
+        'RV': []  # patients with abnormal right ventricle
+    }
+    for data_path in data_paths:
+        with open(data_path, 'r') as f:
+            for line in f:
+                key, value = line.rstrip('\n').split(': ')
+                if key == 'Group':
+                    group_dict[value].append(data_path.parent.resolve())
+                    break
+    _, groups = zip(*sorted(group_dict.items()))
+    random.seed(0)
+    groups = tuple(random.sample(group, k=len(group)) for group in groups)
+    folds = tuple(zip(*groups))  # twenty folds where a fold contains five types of patients.
+
+    output_dir = args.output_dir
+    if not output_dir.is_dir():
+        output_dir.mkdir(parents=True)
+    ratio = len(folds) // (args.k + 1)
```

Comment:
If `args.k` is 5, the dataset will be split into 6 folds (5 for training, 1 for testing)?

Reply:
That's right.

Reply:
In our experiment, we will use the `test_fold` instead of the cases in the testing directory for the verification of our algorithm, right?

Reply:
Yes. But `testing.csv` can still be used if we can upload the results on their website.